### PR TITLE
chore: Added example for how to build container images with different architectures

### PR DESCRIPTION
**Pull request by GonzalezAndrew**
## Description

Added an example to showcase how one could build a container image on a build machine which might have a different architecture.

## Motivation and Context

I struggled to build a docker image using x86_64 architecture with this module when building container images from my local M1 Max mac machine until I found a workaround. #333 

The example I am adding is said work around and could be useful to those who also struggled like myself.

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [X] I have executed `pre-commit run -a` on my pull request

I have tested this example by deploying the example lambda function in my personal AWS account. Additional screenshots demonstrating a deployment can be referenced here #333

### CODE REVIEW

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Looks great! Could you please merge this new example into an existing container image example? The main difference between those is only `Dockerfile`.

Also, please update the description of the example in the parent README.md.

**Comment by GonzalezAndrew** [comment]
Thanks for the review @antonbabenko 

I have done the changes you have requested and have tested the updated container-image example

test

Please let me know if there are any other additional changes that need to be made. Thanks.

File: examples/container-image/context/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM scratch
+FROM --platform=linux/x86_64 scratch
```
Comment (antonbabenko) [inline examples/container-image/context/Dockerfile:1]:
```suggestion
# `--platform` argument is used to be able to build docker images when using another platform (e.g. Apple M1)
FROM --platform=linux/x86_64 scratch
```

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
Thank you @GonzalezAndrew !

**Comment by antonbabenko** [comment]
This PR is included in version 4.0.1 :tada:

**Pull request merged**